doc/user-journey.md

```diff
@@ -88,6 +88,68 @@ The ultimate goal is to eliminate or significantly simplify these steps by enabl
 
 - node/express: TBA
 
-## VSCode
+## TypeSpec Extension for VS Code
 
-TBA
+### Overview of the extension
+
+The extension is focusing on improving the API/SDK producer’s experience in prototyping API using TypeSpec.  It supports IntelliSense for authoring TypeSpec and provides interactive guidance within VS Code IDE for:
+
+- Scaffolding TypeSpec projects using various templates
+- Generating OpenAPI 3.0 from TypeSpec
+- Generating client code from TypeSpec
+- Generating server stub from TypeSpec
+
+You can quickly start and explore TypeSpec’s full capabilities within TypeSpec extension for VS Code.
+
+### Installation
+
+- Install prerequisites
+  - Install node.js 20 LTS or above in an **administrative** shell, verify that `node --version` and `npm --version` run in a command prompt.
+  - Install TypeSpec Compiler in **global** mode.
+
+    ```bash
+    npm install -g @typespec/compiler
+    ```
+
+- Install the extension
+
+Note:
+
+- If TypeSpec Compiler is not installed globally or the version installed is outdated, users will be prompted to install the latest version when the extension is loaded upon opening a .tsp file, opening a TypeSpec project folder, or creating a TypeSpec project.
```

Comment:
1. User can also have compiler installed locally, we won't prompt to install compiler globally in that case
2. The "outdated" may be misleading, it means out-of-service right? not mean "not latest version". I think we dont need to mention it as we don't have OOS scenario now.
3. we will only prompt for installing compiler when creating a typespec project now when compiler is not installed globally or locally.